### PR TITLE
[ASA] Add introduction to algorithms' notes

### PR DESCRIPTION
**Pull request by randomicecube**

### CODE REVIEW

**Comment by randomicecube** [comment]
The request was made so that adding the slides and the frontpage link wasn't forgotten

File: content/asa/0001-introducao.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  $$
+
+Fica, então, provado que $T(n) > fibo(n)$.
+
+:::
+
+Podemos, ainda, referir que:
+
+$$
+\forall_{n \geq 11} \text{ } fibo(n) \geq (\frac{3}{2})^n
```
Comment (diogotcorreia) [inline content/asa/0001-introducao.md:103]:
```suggestion
\forall_{n \geq 11} \text{ } fibo(n) \geq \left(\frac{3}{2}\right)^n
```
Let the parenthesis match the height of the fraction

File: content/asa/0001-introducao.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+Em relação a esta implementação, temos:
+
+$$
+T(n)=\begin{cases}
+c_0 &\text{se } x \leq 1\\
+c_1 \cdot n + c_2 &\text{caso contrário}
+\end{cases}
+$$
+
+O ramo de cima é igual ao da implementação anterior. O de baixo, contudo, é bastante diferente - não estamos dependentes de chamadas recursivas. Temos uma componente $c_1 * n$, correspondente às operações realizadas durante o loop principal, e uma $c_2$ corrrespondente às outras operações da função, todas realizadas em tempo constante.
```
Comment (diogotcorreia) [inline content/asa/0001-introducao.md:147]:
```suggestion
O ramo de cima é igual ao da implementação anterior. O de baixo, contudo, é bastante diferente - não estamos dependentes de chamadas recursivas. Temos uma componente $c_1 \cdot n$, correspondente às operações realizadas durante o loop principal, e uma $c_2$ corrrespondente às outras operações da função, todas realizadas em tempo constante.
```

**Review by diogotcorreia — CHANGES_REQUESTED** [review]

**Review by diogotcorreia — APPROVED** [review]

File: content/asa/0001-introducao.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+sum = \sum^{i - 1}_{k = 0}arr[k]
+$$
+
+onde $i$ é a variável do ciclo que vai sendo incrementada. Podemos verificar, a qualquer momento do loop, que `sum` corresponde, de facto, ao valor daquele somatório.
+
+---
+
+- [Slides](https://drive.google.com/file/d/1iyuzEQdYwsUpZR6HAjMK0jfS2JDtTBON/view?usp=sharing)
+
+- [Notas da Aula - Prof. José Fragoso](https://web.tecnico.ulisboa.pt/jose.fragoso/asa/aula1.pdf)
```
Comment (diogotcorreia) [inline content/asa/0001-introducao.md:209]: Change this to a Google Drive mirror

File: content/asa/0001-introducao.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    sum += arr[i];
+  }
+  return sum;
+}
+```
+
+Aqui, temos que, em [**qualquer momento do loop**](color:orange), a variável `sum` é dada por:
+
+$$
+sum = \sum^{i - 1}_{k = 0}arr[k]
```
Comment (diogotcorreia) [inline content/asa/0001-introducao.md:200]:
```suggestion
\operatorname{sum} = \sum^{i - 1}_{k = 0}\operatorname{arr} [k]
```

**Review by diogotcorreia — CHANGES_REQUESTED** [review]

File: content/asa/exercicios/0001-fichas-2021-2022.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  ::::tab-group
+  :::tab[Diogo Correia]
+  `embed:assets/main.cpp`
+  :::
+  :::tab[Diogo Gaspar]
+  `embed:assets/gaspa.cpp`
+  :::
+  :::tab[Luís Fonseca]
+  `embed:assets/luis.cpp`
+  :::
```
Comment (diogotcorreia) [inline content/asa/exercicios/0001-fichas-2021-2022.md:31]: The file path of these should be more specific, in order to allow to add more C++ solutions in the future.
I'll move them myself, don't worry.

**Review by diogotcorreia — CHANGES_REQUESTED** [review]

**Review by diogotcorreia — APPROVED** [review]

**Pull request merged**